test/Microsoft.Azure.EventHubs.Tests/Client/DiagnosticsTests.cs

```diff
@@ -0,0 +1,379 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.EventHubs.Tests.Client
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+
+   public class DiagnosticsTests : ClientTestBase
```

Comment:
you need to implement IDisposable and dispose listener and subscription after each test, otherwise tests may become pretty unstable because of multiple listeners having different IsEnabled and reacting to the callbacks

Reply:
Thanks, fixed